lib/discover/global.go

```diff
@@ -138,11 +148,16 @@ func (c *globalClient) Lookup(device protocol.DeviceID) (direct []string, relays
 	if resp.StatusCode != 200 {
 		resp.Body.Close()
 		l.Debugln("globalClient.Lookup", qURL, resp.Status)
-		return nil, nil, errors.New(resp.Status)
+		err := errors.New(resp.Status)
+		if secs, _ := strconv.Atoi(resp.Header.Get("Retry-After")); secs > 0 {
```

Comment:
It would be more correct to check the error condition, in case they change what stdlib returns.